### PR TITLE
Convert to website and support slide and Jupyter nb generation

### PR DESCRIPTION
**Pull request by trgardos**
Turns out you can't create revealjs slides from the .qmd files in a quarto book project, but you can from a quarto website project. Go figure! It's actually pretty easy to switch over to the website and the result looks very similar to the book output.

Added a shell script to create slides from each qmd or ipynb file into a _revealjs directory and another script to create Jupyter notebooks from each qmd file into a _jupyter directory.

Updated GH Actions workflow to publish from _site instead of _book.

### CODE REVIEW

File: ds701_book/_quarto.yml [hunk 1]
```diff
@@ -1,51 +1,54 @@
...
-    - part: "Preliminaires"
-      chapters:
+  back-to-top-navigation: true
+  sidebar:
+    style: "docked"
+    search: true
+    contents:
+    - href: index.qmd
+      text: Home
+    - section: "Preliminaires"
```
Comment (sladenheim) [inline ds701_book/_quarto.yml:14]: This is a lingering typo from me that we should fix.

File: ds701_book/cmd-render-slides.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+done
+
+# Render all .ipynb files to HTML and Reveal.js
+for file in *.ipynb; do
+  if [ -f "$file" ]; then
+    quarto render "$file" --to revealjs --output-dir _revealjs
+  fi
+done
+
+# revealjs slideshow
```
Comment (sladenheim) [inline ds701_book/cmd-render-slides.sh:23]: Do we need these commented lines?

**Comment by trgardos** [comment]
> This is great. Nice job. Regarding the convert script, since we currently only have .qmd, the script shouldn't convert any .ipynb files right? Is the intent of the convert script to allow for potential future use (and/or conversion) to .ipynb files?

Yes. But I've been playing with quarto projects and profiles.  it seems it may be possible to do something like `quarto render --profile web` and `quarto render --profile slides` but it is not quite working how I would hope. I'm working on a 2nd PR on top of this one that is the best workaround I found so far. Will explain more there.

File: ds701_book/cmd-render-slides.sh (hunk 2 above)
Comment (trgardos) [reply ds701_book/cmd-render-slides.sh]: removed.

File: ds701_book/_quarto.yml (hunk 1 above)
Comment (trgardos) [reply ds701_book/_quarto.yml]: fixed

**Review by sladenheim — APPROVED** [review]

**Pull request merged**